src/Microsoft.ML.CpuMath/CpuMathUtils.netcoreapp.cs

```diff
@@ -915,11 +939,11 @@ public static void SdcaL1UpdateSparse(float primalUpdate, int count, ReadOnlySpa
             Contracts.Assert(count <= v.Length);
             Contracts.Assert(count <= w.Length);
 
-            if (Avx.IsSupported)
+            if (Avx.IsSupported && count >= MinVectorSize)
```

Comment:
I think, if there is now a `count >= size` checks, the various implementations have some dead-code paths.

Reply:
Actually not. eg if the inputsize is 21 and avx.is supported

It will execute all three code blocks, avx codeblock (0-16), sseblock (16-20) and simple codeblock (21)

Reply:
Some of the functions in `SseIntrinsics.cs` and `AvxIntrinsics.cs` have code paths like:
```cs
if (length < 4)
{
    // Handle cases where we have less than 128-bits total and can't ever use SIMD acceleration.
    switch (length)
    {
        case 3: dst[2] *= scale; goto case 2;
        case 2: dst[1] *= scale; goto case 1;
        case 1: dst[0] *= scale; break;
    }
    return;
}
```

These code paths should be considered dead now.

Reply:
Actually i didnt change those cpumathfunctions which are already using the software implementation for shorter paths. eg scale and sum. but sure i can make them similar to the other cpumathfunctions

Reply:
Ah, I see. I don't think we need to do anything here then.